Source/JavaScriptCore/runtime/JSStringInlines.h

```diff
@@ -138,22 +122,24 @@ class JSStringFibers {
 // Vector before performing any concatenation, but by working backwards we likely
 // only fill the queue with the number of substrings at any given level in a
 // rope-of-ropes.)
-template<typename Fibers, typename CharacterType>
-void JSRopeString::resolveToBufferSlow(Fibers* fibers, CharacterType* buffer, unsigned length)
+template<typename CharacterType>
+void JSRopeString::resolveToBufferSlow(JSString* fiber0, JSString* fiber1, JSString* fiber2, CharacterType* buffer, unsigned length, uint8_t*)
 {
+    // Keep in mind that resolveToBufferSlow signature must be the same to resolveToBuffer to encourage tail-calls by clang, that's the reason why
+    // it takes the last stackLimit parameter still while it is not used here.
+
     CharacterType* position = buffer + length; // We will be working backwards over the rope.
     Vector<JSString*, 32, UnsafeVectorOverflow> workQueue; // These strings are kept alive by the parent rope, so using a Vector is OK.
 
-    for (size_t i = 0; i < JSRopeString::s_maxInternalRopeLength; ++i) {
-        JSString* fiber = fibers->fiber(i);
-        if (!fiber)
-            break;
-        workQueue.append(fiber);
-    }
+    if (fiber0)
+        workQueue.append(fiber0);
+    if (fiber1)
+        workQueue.append(fiber1);
+    if (fiber2)
+        workQueue.append(fiber2);
```

Comment:
Per our offline discussion, this can be the following instead to avoid unnecessary checks:
```
    workQueue.append(fiber0);
    if (fiber1) {
        workQueue.append(fiber1);
        if (fiber2)
            workQueue.append(fiber2);
    }
```

Reply:
Changed.